File: frontend/src/app/shared/model/InstitutionOverviewStat.ts
export interface InstitutionOverviewStat {
    institution: string;
    outgoingStudentsPerHOI: Map<string, number>;
    incomingStudentsPerHOI: Map<string, number>;
    reservationsPerDay: Map<string, number>;
}

export class InstitutionOverviewStatConstructor {
    static new(): InstitutionOverviewStat {
        return {
            institution: '',
            outgoingStudentsPerHOI: new Map<string, number>(),
            incomingStudentsPerHOI: new Map<string, number>(),
            reservationsPerDay: new Map<string, number>(),
        };
    }

    static newFromObj(obj: InstitutionOverviewStat): InstitutionOverviewStat {
        if (obj === null) {
            return null;
        }

        return {
            institution: obj.institution,
            outgoingStudentsPerHOI: obj.outgoingStudentsPerHOI,
            incomingStudentsPerHOI: obj.incomingStudentsPerHOI,
            reservationsPerDay: obj.reservationsPerDay,
        };
    }
}

File: backend/src/main/java/blok2/model/stats/InstitutionOverviewStat.java
package blok2.model.stats;

import javax.persistence.*;
import java.util.Map;

@Entity
public class InstitutionOverviewStat {

    @Id
    private String institution;

    @ElementCollection
    @MapKeyColumn(name = "HOI")
    @Column(name = "count")
    private Map<String, Long> outgoingStudentsPerHOI;

    @ElementCollection
    @MapKeyColumn(name = "HOI")
    @Column(name = "count")
    private Map<String, Long> incomingStudentsPerHOI;

    @ElementCollection
    @MapKeyColumn(name = "date")
    @Column(name = "count")
    private Map<String, Long> reservationsPerDay;

    public InstitutionOverviewStat() {
    }

    public InstitutionOverviewStat(String institution, Map<String, Long> outgoingStudentsPerHOI, Map<String, Long> incomingStudentsPerHOI, Map<String, Long> reservationsPerDay) {
        this.institution = institution;
        this.outgoingStudentsPerHOI = outgoingStudentsPerHOI;
        this.incomingStudentsPerHOI = incomingStudentsPerHOI;
        this.reservationsPerDay = reservationsPerDay;
    }

    public String getInstitution() {
        return institution;
    }

    public Map<String, Long> getOutgoingStudentsPerHOI() {
        return outgoingStudentsPerHOI;
    }

    public Map<String, Long> getIncomingStudentsPerHOI() {
        return incomingStudentsPerHOI;
    }

    public Map<String, Long> getReservationsPerDay() {
        return reservationsPerDay;
    }
}
